dynamoprojection/handler.go

```diff
@@ -0,0 +1,96 @@
+package dynamoprojection
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/dogmatiq/dogma"
+)
+
+// MessageHandler is a specialization of dogma.ProjectionMessageHandler that
+// persists to an AWS DynamoDB database.
+type MessageHandler interface {
+	// Configure produces a configuration for this handler by calling methods on
+	// the configurer c.
+	//
+	// The implementation MUST allow for multiple calls to Configure(). Each
+	// call SHOULD produce the same configuration.
+	//
+	// The engine MUST call Configure() before calling HandleEvent(). It is
+	// RECOMMENDED that the engine only call Configure() once per handler.
+	Configure(c dogma.ProjectionConfigurer)
+
+	// HandleEvent updates the projection to reflect the occurrence of an event.
+	//
+	// It SHOULD return all database changes in the form of the slice to
+	// types.TransactWriteItem that MUST be applied within a single
+	// transaction.
```

Comment:
```suggestion
	// The changes to be made are returned as a slice of transaction items,
	// which MAY be empty. The items are applied to DynamoDB in a single
	// transaction.
```

No need for the `MUST` here, because it's the `adaptor` implementation in this package that handles this, not an unknown engine implementation.